# Guided Capstone Step 4. Pre-Processing and Training Data Development

**The Data Science Method**  


1.   Problem Identification 


2.   Data Wrangling 
  
 
3.   Exploratory Data Analysis   

4.   **Pre-processing and Training Data Development**  
 * Create dummy or indicator features for categorical variables
  * Standardize the magnitude of numeric features
  * Split into testing and training datasets
  * Apply scaler to the testing set
5.   Modeling 
  * Fit Models with Training Data Set
  * Review Model Outcomes — Iterate over additional models as needed.
  * Identify the Final Model

6.   Documentation
  * Review the Results
  * Present and share your findings - storytelling
  * Finalize Code 
  * Finalize Documentation

**<font color='teal'> Start by loading the necessary packages as we did in step 3 and printing out our current working directory just to confirm we are in the correct project directory. </font>**

In [1]:
import os
import numpy as np
import pandas as pd 
import seaborn as sns
import matplotlib.pyplot as plt

os.getcwd()
os.listdir()

['updated_ski_data.csv',
 'step3_revised_data',
 '.DS_Store',
 'fig1.png',
 'GuidedCapstoneStep4 and step5HL.ipynb',
 'step2_output.csv',
 'GuidedCapstoneStep3HL.ipynb',
 'GuidedCapstone_Step2HL.ipynb',
 'GuidedCapstoneStep6HL.ipynb',
 '.ipynb_checkpoints',
 'step4.csv',
 'step3_output.csv',
 '1564524869_Problem_Statement_Worksheet_Template.pptx']

**<font color='teal'>  Load the csv file you created in step 3, remember it should be saved inside your data subfolder and print the first five rows.</font>**

In [7]:
df = pd.read_csv("step3_output.csv")
df = df.drop(['Unnamed: 0','Unnamed: 0.1','Region'],axis=1)
df.head(5)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 176 entries, 0 to 175
Data columns (total 27 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Name               176 non-null    object 
 1   state              176 non-null    object 
 2   summit_elev        176 non-null    int64  
 3   vertical_drop      176 non-null    int64  
 4   base_elev          176 non-null    int64  
 5   trams              176 non-null    int64  
 6   fastEight          176 non-null    float64
 7   fastSixes          176 non-null    int64  
 8   fastQuads          176 non-null    int64  
 9   quad               176 non-null    int64  
 10  triple             176 non-null    int64  
 11  double             176 non-null    int64  
 12  surface            176 non-null    int64  
 13  total_chairs       176 non-null    int64  
 14  Runs               176 non-null    float64
 15  TerrainParks       176 non-null    float64
 16  LongestRun_mi      176 non

## Create dummy features for categorical variables

**<font color='teal'> Create dummy variables for `state`. Add the dummies back to the dataframe and remove the original column for `state`. </font>**

Hint: you can see an example of how to execute this in Aiden's article on preprocessing [here](https://medium.com/@aiden.dataminer/the-data-science-method-dsm-pre-processing-and-training-data-development-fd2d75182967). 

In [8]:
k = pd.concat([df.drop(['state'],axis=1),pd.get_dummies(df['state'])],axis=1)
k.head(3)

Name  summit_elev  vertical_drop  base_elev  \
0               Hilltop Ski Area         2090            294       1796   
1            Sunrise Park Resort        11100           1800       9200   
2  Yosemite Ski & Snowboard Area         7800            600       7200   

   trams  fastEight  fastSixes  fastQuads  quad  triple  ...  Rhode Island  \
0      0        0.0          0          0     0       1  ...             0   
1      0        0.0          0          1     2       3  ...             0   
2      0        0.0          0          0     0       1  ...             0   

   South Dakota  Tennessee  Utah  Vermont  Virginia  Washington  \
0             0          0     0        0         0           0   
1             0          0     0        0         0           0   
2             0          0     0        0         0           0   

   West Virginia  Wisconsin  Wyoming  
0              0          0        0  
1              0          0        0  
2              0          0        0  

[3 rows x 61 columns]

## Standardize the magnitude of numeric features

**<font color='teal'> Using sklearn preprocessing standardize the scale of the features of the dataframe except the name of the resort which we done't need in the dataframe for modeling, so it can be droppped here as well. Also, we want to hold out our response variable(s) so we can have their true values available for model performance review. Let's set `AdultWeekend` to the y variable as our response for scaling and modeling. Later we will go back and consider the `AdultWeekday`, `dayOpenLastYear`, and `projectedDaysOpen`. For now leave them in the development dataframe. </font>**

In [12]:
# first we import the preprocessing package from the sklearn library
from sklearn import preprocessing

# Declare an explanatory variable, called X,and assign it the result of dropping 'Name' and 'AdultWeekend' from the df
#X = k.drop(['Name','AdultWeekend'], axis=1)
X = k.drop(['Name','AdultWeekend','AdultWeekday','clusters'],axis=1)
# Declare a response variable, called y, and assign it the AdultWeekend column of the df 
y = k.AdultWeekend 

# Here we use the StandardScaler() method of the preprocessing package, and then call the fit() method with parameter X 
scaler = preprocessing.StandardScaler().fit(X)

# Declare a variable called X_scaled, and assign it the result of calling the transform() method with parameter X 
X_scaled=scaler.transform(X) 

## Split into training and testing datasets

**<font color='teal'> Using sklearn model selection import train_test_split, and create a 75/25 split with the y = `AdultWeekend`. We will start by using the adult weekend ticket price as our response variable for modeling.</font>**

In [13]:
# Import the train_test_split function from the sklearn.model_selection utility.  
from sklearn.model_selection import train_test_split

# Get the 1-dimensional flattened array of our response variable y by calling the ravel() function on y
y = k.AdultWeekend 
y = k['AdultWeekend'].ravel()

# Call the train_test_split() function with the first two parameters set to X_scaled and y 
# Declare four variables, X_train, X_test, y_train and y_test separated by commas 
X_train,X_test,y_train,y_test= train_test_split(X_scaled, y, test_size=0.25, random_state=1)

In [14]:
X_train

array([[-0.80077675, -0.85212798, -0.71839355, ..., -0.13168538,
        -0.25819889, -0.17099639],
       [-0.61050361,  0.55347569, -0.79288315, ..., -0.13168538,
        -0.25819889, -0.17099639],
       [-0.54151026,  0.19614753, -0.64251809, ..., -0.13168538,
        -0.25819889, -0.17099639],
       ...,
       [-0.80232369, -0.9368029 , -0.7028027 , ..., -0.13168538,
        -0.25819889, -0.17099639],
       [ 1.51808048,  1.26474502,  1.52564902, ..., -0.13168538,
        -0.25819889, -0.17099639],
       [-0.81500856, -0.64044068, -0.77763877, ..., -0.13168538,
        -0.25819889, -0.17099639]])

Here we start the actual modeling work. First let's fit a multiple linear regression model to predict the `AdultWeekend` price.

# Guided Capstone Step 5. Modeling


This is the fifth step in the Data Science Method. In the previous steps you cleaned and prepared the datasets. Now it's time to get into the most exciting part: modeling! In this exercise, you'll build three different models and compare each model's performance. In the end, you'll choose the best model for demonstrating insights to Big Mountain management.



### **The Data Science Method**  


1.   Problem Identification 

2.   Data Wrangling 
  
3.   Exploratory Data Analysis 
 
4.   Pre-processing and Training Data Development

5.   **Modeling**
  * Fit Models with Training Data Set
  * Review Model Outcomes — Iterate over additional models as needed.
  * Identify the Final Model

6.   Documentation
  * Review the Results
  * Present and share your findings - storytelling
  * Finalize Code 
  * Finalize Documentation

## Fit Models with a Training Dataset

**<font color='teal'> Using sklearn, fit the model on your training dataset.</font>**

In [18]:
!pip3 install statsmodels

    100% |████████████████████████████████| 8.4MB 4.0MB/s eta 0:00:01
    100% |████████████████████████████████| 235kB 16.3MB/s ta 0:00:01
You are using pip version 19.0.3, however version 20.2b1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.


#### Model 1

In [15]:
#all first model set
# from sklearn import linear_model
# from sklearn.metrics import explained_variance_score,mean_absolute_error
# lm = linear_model.LinearRegression()
# model = lm.fit(X_train,y_train)
import statsmodels.api as sm
X_train = sm.add_constant(X_train)
model = sm.OLS(y_train,X_train)
model = model.fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.807
Model:                            OLS   Adj. R-squared:                  0.683
Method:                 Least Squares   F-statistic:                     6.545
Date:                Wed, 01 Jul 2020   Prob (F-statistic):           6.91e-14
Time:                        21:41:01   Log-Likelihood:                -442.63
No. Observations:                 132   AIC:                             989.3
Df Residuals:                      80   BIC:                             1139.
Df Model:                          51                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
x1            94.2174     84.907      1.110      0.270     -74.753     263.187
x2           -11.6604     15.803     -0.738      0.463     -43.110      19.790
x3           -80.7448     74.999     -1.077      0.285    -229.998      68.508
const      -5.036e-13   1.06e-12     -0.476      0.635   -2.61e-12     1.6e-12
x4          9.767e-13   3.12e-13      3.133      0.002    3.56e-13     1.6e-12
x5          1.558e-12   1.18e-12      1.323      0.190   -7.85e-13     3.9e-12
x6             0.5707      1.218      0.469      0.641      -1.853       2.994
x7             2.9811      1.052      2.834      0.006       0.887       5.075
x8             2.6900      1.109      2.426      0.018       0.483       4.897
x9             1.2345      1.060      1.165      0.248      -0.875       3.344
x10           -1.5464      0.969     -1.596      0.114      -3.474       0.382
x11            1.9545      0.702      2.783      0.007       0.557       3.352
x12            2.0057      2.318      0.865      0.390      -2.608       6.619
x13            0.1718      1.235      0.139      0.890      -2.287       2.630
x14            1.7278      1.386      1.246      0.216      -1.031       4.486
x15           -2.1932      1.698     -1.292      0.200      -5.571       1.185
x16            2.1571      1.365      1.581      0.118      -0.559       4.873
x17            0.8950      1.489      0.601      0.549      -2.067       3.857
x18           -1.4932      1.156     -1.292      0.200      -3.794       0.807
x19            2.3673      1.827      1.295      0.199      -1.269       6.004
x20           -1.4881      1.470     -1.012      0.314      -4.413       1.437
x21            0.9838      1.140      0.863      0.391      -1.284       3.252
x22            4.0032      0.759      5.276      0.000       2.493       5.513
x23            3.2474      1.134      2.864      0.005       0.991       5.504
x24           13.2382      1.947      6.800      0.000       9.364      17.113
x25           10.7323      1.742      6.160      0.000       7.265      14.200
x26           12.0883      1.097     11.020      0.000       9.905      14.271
x27            7.9643      1.328      5.999      0.000       5.322      10.607
x28            7.5901      1.045      7.260      0.000       5.510       9.671
x29            5.4874      0.883      6.212      0.000       3.729       7.245
x30            7.4103      1.049      7.064      0.000       5.323       9.498
x31            8.8264      1.358      6.499      0.000       6.123      11.529
x32            5.0356      0.769      6.546      0.000       3.505       6.566
x33            8.9900      0.961      9.360      0.000       7.079      10.902
x34           16.4284      1.306     12.581      0.000      13.830      19.027
x35           11.8145      1.021     11.577      0.000       9.784      13.845
x3

**<font color='teal'> Predict on the testing dataset and score the model performance with the y_test set and the y-pred values. The explained variance is a measure of the variation explained by the model. This is also known as the R-squared value. </font>**

Hint: you will have to use the `predict()` method here as it's used in this [DSM article](https://medium.com/@aiden.dataminer/the-data-science-method-dsm-modeling-56b4233cad1b) about modeling. 

In [369]:
# Make a variable called y_pred and assign it the result of calling predict() on our model variable with parameter X_test\
y_pred =model.predict(X_test)

## Review Model Outcomes — Iterate over additional models as needed

In [176]:
# You might want to use the explained_variance_score() and mean_absolute_error() metrics.
# To do so, you will need to import them from sklearn.metrics. 
# You can plug y_test and y_pred into the functions to evaluate the model

In [370]:
from sklearn.metrics import explained_variance_score, mean_absolute_error
explained_variance_score(y_test,y_pred), mean_absolute_error(y_test,y_pred)

(-11706.639930117824, 231.7118369096451)

**<font color='teal'> Print the intercept value from the linear model. </font>**

Hint: our linear regression model `lm` has an attribute `intercept_` for the intercept

In [371]:
model.params[3]

5.460818146023717e-13

**<font color='teal'> The intercept is the mean `AdultWeekend` price for all the resorts given the other characteristics. The addition or subtraction of each of the coefficient values in the regression are numeric adjustments applied to the intercept to provide a particular observation's value for the resulting `AdultWeekend` value. Also, because we took the time to scale our x values in the training data, we can compare each of the coeeficients for the features to determine the feature importances. Print the coefficient values from the linear model and sort in descending order to identify the top ten most important features.</font>** 


Hint: make sure to review the absolute value of the coefficients, because the adjustment may be positive or negative, but what we are looking for is the magnitude of impact on our response variable.

In [372]:
# You might want to make a pandas DataFrame displaying the coefficients for each state like so: 
df1 = pd.DataFrame(abs(model.params), X.columns,columns=['Coefficient'])
df1.sort_values(by = 'Coefficient',ascending=False).head(10)

Coefficient
New Jersey      738.807029
summit_elev      29.526120
base_elev        26.798305
New York         19.555365
Michigan         16.277611
New Hampshire    15.886130
Pennsylvania     15.184512
Wisconsin        14.662033
California       13.915831
Vermont          12.593484

**<font color='teal'>You should see that the top ten important features are different states. However, the state is not something the managers at the Big Mountain Resort can do anything about. Given that we care more about actionable traits associated with ticket pricing, rebuild the model without the state features and compare the results. </font>**

Hint: Try to construct another model using exactly the steps we followed above. 

#### Model 2

In [373]:
lst = df1.sort_values(by = 'Coefficient',ascending=False).head(10).index
lst

Index(['New Jersey', 'summit_elev', 'base_elev', 'New York', 'Michigan',
       'New Hampshire', 'Pennsylvania', 'Wisconsin', 'California', 'Vermont'],
      dtype='object')

In [470]:
k = pd.concat([df.drop(['state'],axis=1),pd.get_dummies(df['state'])],axis=1)
X = k.drop(['New Jersey', 'New York', 'Michigan', 'New Hampshire', 'Pennsylvania',
       'Wisconsin', 'California', 'Vermont','Name','AdultWeekend'],axis=1)
scaler = preprocessing.StandardScaler().fit(X)
X_scaled=scaler.transform(X) 
y = k.AdultWeekend 
y = k['AdultWeekend'].ravel()

In [471]:
X_train,X_test,y_train,y_test = train_test_split(X_scaled,y, test_size=0.25,random_state=12)

In [472]:
import statsmodels.api as sm
X_train = sm.add_constant(X_train)
model = sm.OLS(y_train,X_train)
model = model.fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.868
Model:                            OLS   Adj. R-squared:                  0.800
Method:                 Least Squares   F-statistic:                     12.61
Date:                Sun, 21 Jun 2020   Prob (F-statistic):           4.92e-23
Time:                        20:07:00   Log-Likelihood:                -411.39
No. Observations:                 132   AIC:                             914.8
Df Residuals:                      86   BIC:                             1047.
Df Model:                          45                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
x1           -30.7265     52.926     -0.581      0.563    -135.940      74.487
x2             9.7037      9.858      0.984      0.328      -9.893      29.300
x3            21.5869     47.274      0.457      0.649     -72.391     115.565
const      -2.555e-13   1.83e-14    -13.981      0.000   -2.92e-13   -2.19e-13
x4          4.964e-13   2.72e-14     18.224      0.000    4.42e-13    5.51e-13
x5         -4.458e-13   9.87e-15    -45.192      0.000   -4.65e-13   -4.26e-13
x6             1.3058      0.921      1.417      0.160      -0.526       3.137
x7             1.6902      0.797      2.120      0.037       0.106       3.275
x8             1.4801      0.803      1.844      0.069      -0.116       3.076
x9             0.5496      0.663      0.829      0.409      -0.768       1.867
x10           -1.4534      0.690     -2.107      0.038      -2.825      -0.082
x11            0.9121      0.455      2.004      0.048       0.007       1.817
x12            2.1433      1.476      1.452      0.150      -0.790       5.077
x13            0.4348      0.815      0.533      0.595      -1.186       2.056
x14            1.2436      0.989      1.257      0.212      -0.723       3.211
x15           -3.1139      1.365     -2.281      0.025      -5.827      -0.400
x16           -0.3758      1.024     -0.367      0.715      -2.411       1.660
x17            0.2359      1.068      0.221      0.826      -1.888       2.360
x18            0.8231      0.789      1.044      0.300      -0.745       2.391
x19            0.1683      1.295      0.130      0.897      -2.406       2.743
x20            8.3836      1.071      7.825      0.000       6.254      10.513
x21           -1.5498      1.195     -1.297      0.198      -3.924       0.825
x22            0.0996      0.761      0.131      0.896      -1.414       1.613
x23            8.1714      4.689      1.743      0.085      -1.150      17.492
x24           -0.0729      0.587     -0.124      0.901      -1.240       1.094
x25         -374.6468      4.649    -80.581      0.000    -383.889    -365.404
x26           -3.3680      1.922     -1.753      0.083      -7.188       0.452
x27            0.6434      0.757      0.850      0.398      -0.862       2.148
x28           -2.3275      0.981     -2.372      0.020      -4.278      -0.377
x29           -0.6657      0.588     -1.133      0.260      -1.834       0.502
x30           -0.9570      0.585     -1.635      0.106      -2.120       0.206
x31           -0.5266      0.581     -0.907      0.367      -1.681       0.628
x32           -1.7226      1.258     -1.370      0.174      -4.223       0.777
x33            0.8720      0.596      1.463      0.147      -0.313       2.057
x34            0.4769      0.774      0.616      0.539      -1.061       2.015
x35           -0.7411      0.613     -1.209      0.230      -1.960       0.478
x3

In [473]:
y_pred =model.predict(X_test)
explained_variance_score(y_test,y_pred), mean_absolute_error(y_test,y_pred)

(-5013.416582311873, 234.56700108921552)

In [474]:
explained_variance_score(y_test,y_pred), mean_absolute_error(y_test,y_pred)
df1 = pd.DataFrame(abs(model.params), X.columns,columns=['Coefficient'])
df1.sort_values(by = 'Coefficient',ascending=False).head(10)

Coefficient
Arizona             374.646823
Nevada              374.646823
summit_elev          30.726548
base_elev            21.586860
vertical_drop         9.703731
AdultWeekday          8.383618
clusters              8.171367
Colorado              3.368048
New Mexico            3.297554
SkiableTerrain_ac     3.113892

**<font color='teal'> When reviewing our new model coefficients, we see `summit_elev` is now in the number two spot. This is also difficult to change from a management prespective and highly correlated with `base_elev` and `vertical_drop`.  This time, rebuild the model without the state features and without the `summit_elev` and without `base_elev`and compare the results. </font>**

#### Model 3

In [34]:
X1 = X.drop(['summit_elev','base_elev'],axis=1)
scaler = preprocessing.StandardScaler().fit(X1)
X_scaled=scaler.transform(X1) 
X_train,X_test,y_train,y_test = train_test_split(X_scaled,y,test_size=0.25,random_state=1)
model = lm.fit(X_train,y_train)
df2 = pd.DataFrame(abs(model.coef_),X1.columns,columns=['Coefficient'])
df2.sort_values(by = 'Coefficient',ascending=False).head(10)

Coefficient
vertical_drop         8.309728
Snow Making_ac        6.175149
SkiableTerrain_ac     5.460195
Runs                  5.400427
fastQuads             4.174327
quad                  3.591693
trams                 3.095555
triple                3.091791
surface               3.022515
daysOpenLastYear      2.821945

In [33]:
y_pred =model.predict(X_test)
explained_variance_score(y_test,y_pred), mean_absolute_error(y_test,y_pred)

(0.3048384086313497, 14.345420066684458)

## Identify the Final Model

**<font color='teal'> Review the model performances in the table below and choose the best model for proving insights to Big Mountain management about what features are driving ski resort lift ticket prices. Type your choice in the final markdown cell — you will discuss this selection more in the next step of the guided casptone. </font>**

| Model | Explained Variance| Mean Absolute Error|Features Dropped|
| --- | --- | --- | --- |
| Model 1. | -5599.51| 269.799 |-|
| Model 2. | 0.3302| 13.1015|'state'|
| Model 3. | 0.3048| 14.3542|'state','summit_elev','base_elev'|

Model Selection:

In [21]:
from sklearn import linear_model
from sklearn.metrics import explained_variance_score,mean_absolute_error
df = pd.read_csv('step3_revised_data',index_col=0)
lm = linear_model.LinearRegression()
k = pd.concat([df.drop(['state'],axis=1),pd.get_dummies(df['state'])],axis=1)
X = k.drop(['Name','AdultWeekend','AdultWeekday'],axis=1)
scaler = preprocessing.StandardScaler().fit(X)
X_scaled=scaler.transform(X) 
y = k.AdultWeekend 
y = k['AdultWeekend'].ravel()
X_train,X_test,y_train,y_test = train_test_split(X_scaled,y, test_size=0.25,random_state=12)
X_train = sm.add_constant(X_train)
model = sm.OLS(y_train,X_train)
model = model.fit()
df2 = pd.DataFrame(abs(model.params),X.columns,columns=['Coefficient'])
df2.sort_values(by = 'Coefficient',ascending=False).head(10)

Coefficient
Tennessee       586.718899
fastEight       586.718899
vertical_drop    10.660132
Utah             10.576591
Runs             10.468323
New York         10.211580
summit_elev      10.116384
California        9.884108
Pennsylvania      9.788015
Michigan          9.272432

In [22]:
y_pred =model.predict(X_test)
explained_variance_score(y_test,y_pred), mean_absolute_error(y_test,y_pred)

(-5599.38989534983, 269.9123053756259)

In [30]:
#k = pd.concat([df.drop(['state'],axis=1),pd.get_dummies(df['state'])],axis=1)
# X = k.drop(['New Jersey', 'New York', 'Michigan', 'New Hampshire', 'Pennsylvania',
#        'Wisconsin', 'California','Arizona','Nevada','Name','AdultWeekend'],axis=1)
# X = k.drop(['Tennessee', 'New York', 'Michigan', 'California', 'Pennsylvania','Utah',
#             'Name','AdultWeekend','AdultWeekday'],axis=1)
X = df.drop(['Name','AdultWeekend','AdultWeekday','state'],axis=1)
scaler = preprocessing.StandardScaler().fit(X)
X_scaled=scaler.transform(X) 
y = k.AdultWeekend 
y = k['AdultWeekend'].ravel()
X_train,X_test,y_train,y_test = train_test_split(X_scaled,y, test_size=0.25,random_state=12)
# X_train = sm.add_constant(X_train)
# model = sm.OLS(y_train,X_train)
model = lm.fit(X_train,y_train)
df2 = pd.DataFrame(abs(model.coef_),X.columns,columns=['Coefficient'])
df2.sort_values(by = 'Coefficient',ascending=False).head(10)

Coefficient
Runs                  8.747491
SkiableTerrain_ac     7.821495
summit_elev           6.339789
base_elev             5.621068
fastQuads             5.084209
vertical_drop         4.553922
triple                3.514875
surface               3.405169
quad                  2.766515
total_chairs          2.171289

In [31]:
y_pred =model.predict(X_test)
explained_variance_score(y_test,y_pred), mean_absolute_error(y_test,y_pred)

(0.33019374360608866, 13.10154378976526)

In [26]:
X = X.drop(['summit_elev','base_elev'],axis=1)
scaler = preprocessing.StandardScaler().fit(X)
X_scaled=scaler.transform(X) 
X_train,X_test,y_train,y_test = train_test_split(X_scaled,y,test_size=0.25,random_state=1)
model = lm.fit(X_train,y_train)
df2 = pd.DataFrame(abs(model.coef_),X.columns,columns=['Coefficient'])
df2.sort_values(by = 'Coefficient',ascending=False).head(10)

Coefficient
vertical_drop         8.309728
Snow Making_ac        6.175149
SkiableTerrain_ac     5.460195
Runs                  5.400427
fastQuads             4.174327
quad                  3.591693
trams                 3.095555
triple                3.091791
surface               3.022515
daysOpenLastYear      2.821945

In [27]:
y_pred =model.predict(X_test)
explained_variance_score(y_test,y_pred), mean_absolute_error(y_test,y_pred)

(0.3048384086313497, 14.345420066684458)

In [615]:
df.to_csv('step4.csv')